app/controllers/api/subcollections/container_images.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Api
+  module Subcollections
+    module ContainerImages
+      def container_images_scan_resource(provider_id, type, image_id, _payload)
+        api_action(type, image_id, :skip_href => true) do |klass|
+          image = resource_search(image_id, type, klass)
+          begin
+            task_id = image.scan
+            if task_id.present?
+              desc = "#{container_image_ident(image)} scanning"
+              action_result(true, desc).merge(:task_id => task_id, :parent_id => provider_id)
+            else
+              desc = "#{container_image_ident(image)} failed to start scanning"
```

Comment:
Is there any way that we can bubble up to the user what the problem was?

Reply:
This `image.scan` method only returns a task or nil. In the specific case of not defined MiqEventDefinition it will write to the evm log a warning but I don't see how to catch it here. I am hoping that if there will be a problem it will raise an exception.